src/formatDate.js

```diff
@@ -8,7 +8,60 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
-}
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+
+  const dateParts = date.split(fromSeparator);
+  const newDate = [];
+  const newDateObj = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (
+      fromFormat[i] === 'YYYY' ||
+      fromFormat[i] === 'YY' ||
+      fromFormat[i] === 'MM' ||
+      fromFormat[i] === 'DD'
+    ) {
+      newDateObj[fromFormat[i]] = dateParts[i];
+    }
+  }
+
+  const centuryThreshold = 30;
+  const pastCentury = '19';
+  const actualCentury = '20';
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    newDateObj['YY'] = newDateObj['YYYY'].slice(-2);
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (newDateObj['YY'] < centuryThreshold) {
+      newDateObj['YYYY'] = actualCentury + newDateObj['YY'];
+    } else {
+      newDateObj['YYYY'] = pastCentury + newDateObj['YY'];
+    }
+  }
 
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const part = toFormat[i];
+
+    if (part === 'YYYY') {
+      newDate.push(newDateObj['YYYY']);
+    }
+
+    if (part === 'YY') {
+      newDate.push(newDateObj['YY']);
+    }
+
+    if (part === 'MM') {
+      newDate.push(newDateObj['MM']);
+    }
+
+    if (part === 'DD') {
+      newDate.push(newDateObj['DD']);
+    }
+  }
```

Comment:
The loop that constructs the new date string is redundant. You can map over the 'toFormat' array, transform each part accordingly, and then join them with the 'toSeparator'. This would help to avoid repetitive if statements and make the code more concise.